### ISSUE TITLE
Unable to package a python code using nuitka

### ISSUE THREAD

**Issue by animeshgupta720** [Nuitka/Nuitka#2765]
- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

```(virtual_env_drit) agupta4@C02C124WMD6P packaged_app % python -m nuitka --version
2.2rc2
Commercial: None
Python: 3.9.18 (main, Mar  8 2024, 08:43:14)
Flavor: pyenv
Executable: /Users/agupta4/.pyenv/versions/3.9.18/bin/python
OS: Darwin
Arch: x86_64
Version C compiler: /usr/bin/clang (clang 15.0.0).
```
- How did you install Nuitka and Python

  I installed with pip

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip freeze

`(virtual_env_drit) agupta4@C02C124WMD6P packaged_app % python -m pip freeze
altgraph==0.17
importlib-metadata==4.6.1
macholib==1.14
Nuitka @ https://github.com/Nuitka/Nuitka/archive/factory.zip
ordered-set==4.1.0
pyinstaller==4.6
pyinstaller-hooks-contrib==2021.2
PyQt5==5.15.4
PyQt5-Qt5==5.15.2
PyQt5-sip==12.9.0
PySide2==5.15.2
python_http_client==3.3.2
sendgrid==6.7.1
shiboken2==5.15.2
starkbank-ecdsa==1.1.1
typing-extensions==3.10.0.0
zipp==3.5.0
zstandard==0.22.0`

This is my error:

`(virtual_env_drit) agupta4@C02C124WMD6P packaged_app % python -m nuitka --follow-imports ~/Documents/Neteng-Dr-IT/lib/drIT/main.py
Nuitka-Options: Used command line options: --follow-imports /Users/agupta4/Documents/Neteng-Dr-IT/lib/drIT/main.py
Nuitka: Starting Python compilation with Nuitka '2.2rc2' on Python '3.9' commercial grade 'not installed'.
Nuitka: Completed Python level compilation and optimization.
Nuitka: Generating source code for C backend compiler.
Nuitka: Running data composer tool for optimal constant value handling.
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: clang (clang 15.0.0).
Nuitka-Scons: Backend linking program with 110 files (no progress information available for this stage).
ld: Undefined symbols:
  _libintl_bind_textdomain_codeset, referenced from:
      _PyIntl_bind_textdomain_codeset in libpython3.9.a[135](_localemodule.o)
  _libintl_bindtextdomain, referenced from:
      _PyIntl_bindtextdomain in libpython3.9.a[135](_localemodule.o)
      _PyIntl_bindtextdomain in libpython3.9.a[135](_localemodule.o)
  _libintl_dcgettext, referenced from:
      _PyIntl_dcgettext in libpython3.9.a[135](_localemodule.o)
  _libintl_dgettext, referenced from:
      _PyIntl_dgettext in libpython3.9.a[135](_localemodule.o)
  _libintl_gettext, referenced from:
      _PyIntl_gettext in libpython3.9.a[135](_localemodule.o)
  _libintl_setlocale, referenced from:
      _PyLocale_setlocale in libpython3.9.a[135](_localemodule.o)
      _PyLocale_setlocale in libpython3.9.a[135](_localemodule.o)
      _PyLocale_localeconv in libpython3.9.a[135](_localemodule.o)
      _PyLocale_localeconv in libpython3.9.a[135](_localemodule.o)
      _PyLocale_localeconv in libpython3.9.a[135](_localemodule.o)
      _PyLocale_localeconv in libpython3.9.a[135](_localemodule.o)
  _libintl_textdomain, referenced from:
      _PyIntl_textdomain in libpython3.9.a[135](_localemodule.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [/Users/agupta4/virtual_env_drit/packaged_app/main.bin] Error 1
FATAL: Failed unexpectedly in Scons C backend compilation.
Nuitka:WARNING:     Complex topic! More information can be found at https://nuitka.net/info/scons-backend-failure.html
Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'.`

**Comment by animeshgupta720** [comment]
Hi all, I am unable to package app using nuitko and getting the error shared above, can someone please be kind enough to look into this and help me out?
I am running the below to package my app,
**python -m nuitka --follow-imports ~/Documents/Neteng-Dr-IT/lib/drIT/main.py** 
The app runs fine locally. Also, it gets packaged with pyinstaller but not with nuitko.

**Comment by kayhayen** [comment]
The flavor pyenv is not working well, certainly not on macOS, please use CPython official download (recommended), Anaconda or Homebrew, those work well with Nuitka.

**Comment by kayhayen** [comment]
If anybody wants to ever make pyenv work with Nuitka on macOS (I think Linux works), and/or provide instructions to users how to do it, be my guest, but it appears to be very broken currently.

**Comment by u-tung** [comment]
The shims method of pyenv may have caused some errors. I replaced `python -m nuitka` with `$(pyenv which nuitka3)`, and it works fine now!
